tree_unique_args/tests/interpreter.rs

```diff
@@ -182,6 +193,15 @@ pub fn interpret(e: &Expr, arg: &Option<Value>, vm: &mut VirtualMachine) -> Valu
             };
             Value::Num(n1 - n2)
         }
+        Expr::Mul(e1, e2) => {
+            let Value::Num(n1) = interpret(e1, arg, vm) else {
+                panic!("mul")
+            };
+            let Value::Num(n2) = interpret(e2, arg, vm) else {
```

Comment:
I think you can leave off the else cases since they should be typechecked. I think rust just raises an error in the else by default

Reply:
a) Ryan put those there
b) Pretty sure it doesn't compile without them